zarr/_storage/store.py

```diff
@@ -22,9 +23,21 @@
 DEFAULT_ZARR_VERSION = 2
 
 v3_api_available = os.environ.get("ZARR_V3_EXPERIMENTAL_API", "0").lower() not in ["0", "false"]
+_has_warned_about_v3 = False  # to avoid printing the warning multiple times
```

Comment:
I thought this was the default but 🤷🏽 

https://docs.python.org/3/library/warnings.html#the-warnings-filter

Reply:
I certainly tried this first. It's possible this would have worked if I had put warning elsewhere, but as implemented, the warning is emitted multiple times without this switch. I think this is due to the fact that `assert_zarr_v3_api_available` is used in various places around the library.